dfastmi/RiversObject.py

```diff
@@ -0,0 +1,569 @@
+from packaging import version
+import configparser
+from typing import Tuple, List, TypedDict, Optional
+import zlib
+from dfastmi.io import collect_int_values1, collect_values_logical, log_text 
+
+class RiversObject(TypedDict, total=False):
+    version: str
+    branches: List[str]
+    reaches: List[List[str]]
+    qlocations: List[str]
+    normal_width: List[List[float]]
+    ucritical: List[List[float]]
+    qstagnant: List[List[float]]
+    tide: List[List[bool]]
+    hydro_q: List[List[Tuple[float, ...]]]
+    autotime: List[List[bool]]
+    hydro_t: List[List[Tuple[float, ...]]]
+    qfit: List[List[Tuple[float, float]]]
+    tide_bc: List[List[Tuple[float, ...]]]
+    cform: List[List[int]]
+    cdisch: List[List[Tuple[float, float]]]
+    prop_q: List[List[Tuple[float, ...]]]
+    prop_c: List[List[Tuple[float, ...]]]
+    proprate_high: List[List[float]] # only for version = 1
+    proprate_low: List[List[float]] # only for version = 1
+    qbankfull: List[List[float]] # only for version = 1
+    qmin: List[List[float]] # only for version = 1
+    qlevels: List[List[Tuple[float, float, float, float]]] # only for version = 1
+    dq: List[List[Tuple[float, float]]] # only for version = 1
+
+def read_rivers(filename: str = "rivers.ini") -> RiversObject:
```

Comment:
From cmd.py a call to this method is made, but it is incorrect, this needs to be fixed, see:
"rivers = dfastmi.io.read_rivers(abs_rivers_file)"

Reply:
done, good find thanks!